src/lib/dev-environment/dev-environment-cli.js

```diff
@@ -53,3 +54,153 @@ export function printTable( data: Object ) {
 
 	console.log( formattedData );
 }
+
+export function processComponentOptionInput( passedParam: string, type: string ) {
+	// cast to string
+	const param = passedParam + '';
+	if ( param.includes( '/' ) ) {
+		return {
+			mode: 'local',
+			dir: param,
+		};
+	}
+
+	if ( type === 'jetpack' && param === 'mu' ) {
+		return {
+			mode: 'inherit',
+		};
+	}
+
+	return {
+		mode: 'image',
+		image: DEV_ENVIRONMENT_CONTAINER_IMAGES[ type ].image,
+		tag: param,
+	};
+}
+
+type NewInstanceOptions = {
+	title: string,
+	multisite: boolean,
+	phpVersion: string,
+	wordpress: string,
+	muPlugins: string,
+	jetpack: string,
+	clientCode: string
+}
+
+const promptIntro = 'This is a wizard to help you set up you local dev environment.\n\n' +
+	'Sensible defaualt values were pre-selected for convinience. ' +
+	'You can also choose to create multiple different environments with different settings using the --slug option.\n\n';
+
+export async function promptForArguments( providedOptions: NewInstanceOptions ) {
+	console.log( promptIntro );
+
+	const instanceData = {
+		wpTitle: providedOptions.title || await promptForText( 'WordPress site title', DEV_ENVIRONMENT_DEFAULTS.title ),
+		phpVersion: providedOptions.phpVersion || await promptForText( 'Php version', DEV_ENVIRONMENT_DEFAULTS.phpVersion ),
+		multisite: providedOptions.multisite || await promptForBoolean( 'Multisite', DEV_ENVIRONMENT_DEFAULTS.multisite ),
+		wordpress: {},
+		muPlugins: {},
+		jetpack: {},
+		clientCode: {},
+	};
+
+	const components = [ 'wordpress', 'muPlugins', 'jetpack', 'clientCode' ];
```

Comment:
If this doesn't change, we can extract it from the function so we don't evaluate it each time. As the question above, not sure if leave it in this file or even in the constants file.

Reply:
I am not sure about this one. The values in that array are kinda referenced in the `promptForComponent` function a lot so I kinda like that they are here close to each other.

What would be the benefit of moving it out of the function? I guess it would only get evaluated at a slightly different time (when the script requires this file as oppose to when it runs the function). But this is not something that would impact performance I assume, or?

What we could do is to move the individual components as a constants and constract an array of constants and use that in the code bellow as well. What do you think?

Reply:
I think the performance benefit would be minimal (especially given that this is a CLI and will only be executed once anyway), so feel free to dismiss my point.